cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/oidc/OIDCUtil.java

```diff
@@ -171,46 +173,126 @@ private static Key getPrivateKey() {
+        if (config == null || keys.isEmpty())
+        {
+            throw new RuntimeException("failed to read any OIDC property ");
+        }
+        
+        Set<String> clientIDs = getClientIDs(config);
+        for (String clientID : clientIDs) {
+            final String secret = getSecret(config, keys, clientID);
+            final String description = getDescription(config, keys, clientID);
+            final boolean signDocuments = getSignDocuments(config, keys, clientID);
+            List<Claim> claims = getClaims(config, keys, clientID);
+
+            RelyParty relyParty = new RelyParty(clientID, secret, description, claims, signDocuments);
             
-            Set<String> clientIDs = config.keySet();
-            for (String c : clientIDs) {
-                String clientSecret = config.getProperty(c).get(0);
-                
-                // TODO: Alinga
-                // these four fields (plus the groupID field) will be moved to the properties file
-                // in the next version of this class.
-                String description = null;
-                List<Claim> claims = null;
-                boolean signDocuments = false;
-                if (c.equals("arbutus-harbor")) {
-                    description = "CANFAR Image Repository";
-                    claims = Arrays.asList(new Claim[] {RelyParty.Claim.NAME, RelyParty.Claim.EMAIL, RelyParty.Claim.GROUPS});
-                    signDocuments = true;
-                }
-                if (c.equals("unions_wiki")) {
-                    description = "UNIONS Wiki";
-                    claims = Arrays.asList(new Claim[] {RelyParty.Claim.EMAIL});
-                    signDocuments = false;
+            // access group is optional
+            String accessGroupKey = clientID + ".access-group";
+            if (keys.contains(accessGroupKey)) {
+                final String accessGroupString = config.getProperty(accessGroupKey).get(0);
+                if (!accessGroupString.isEmpty()) {
+                    GroupURI accessGroup = new GroupURI(URI.create(accessGroupString));
+                    relyParty.setAccessGroup(accessGroup);
                 }
-                relyParties.put(c, new RelyParty(c, clientSecret, description, claims, signDocuments));
             }
+
+            relyParties.put(clientID, relyParty);
+        }
+
+    }
+    
+    public static RelyParty getRelyParty(String clientID) {
+        if (relyParties == null) {
+            // add all rely parties
+            loadConfig();
         }
 
         return relyParties.get(clientID);
     }
     
+    public static boolean accessAllowed(String clientID) {
+        RelyParty rp = getRelyParty(clientID);
+        GroupURI accessGroup = rp.getAccessGroup();
+        if (accessGroup == null) {
+            // access group not specified, allow access
+            return true;
+        } else {
+            GMSClient gmsClient = new GMSClient(accessGroup.getServiceID());
```

Comment:
Instead of using GMSClient, use GroupUtil.getGroupClient() which will return a GroupClient

Reply:
Okay.